src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -961,83 +1009,122 @@ codet java_string_library_preprocesst::make_float_to_string_code(
 
   // Declaring and allocating String * str
   exprt str=allocate_fresh_string(type.return_type(), loc, symbol_table, code);
-  exprt tmp_string=fresh_string(type.return_type(), loc, symbol_table);
 
-  // Declaring CPROVER_string string_expr
-  string_exprt string_expr=fresh_string_expr(loc, symbol_table, code);
-  exprt string_expr_sym=fresh_string_expr_symbol(loc, symbol_table, code);
-
-  // List of the different cases
-  std::vector<code_ifthenelset> case_list;
-
-  // First case in the list is the default
-  code_ifthenelset case_sci_notation;
+  // Expression representing 0.0
   ieee_float_spect float_spec=ieee_float_spect::single_precision();
-  // Subcase of 0.0
-  // TODO: case of -0.0
   ieee_floatt zero_float(float_spec);
   zero_float.from_float(0.0);
   constant_exprt zero=zero_float.to_expr();
-  case_sci_notation.cond()=ieee_float_equal_exprt(arg, zero);
-  case_sci_notation.then_case()=code_assign_string_literal_to_string_expr(
-    string_expr, tmp_string, "0.0", symbol_table);
 
-  // Subcase of computerized scientific notation
-  case_sci_notation.else_case()=code_assign_function_to_string_expr(
-    string_expr, ID_cprover_string_of_float_func, {arg}, symbol_table);
-  case_list.push_back(case_sci_notation);
+  // For each possible case with have a condition and a string_exprt
+  std::vector<exprt> condition_list;
+  std::vector<string_exprt> string_expr_list;
+
+  // Case of computerized scientific notation
```

Comment:
I used the same terminology as the Java documentation https://docs.oracle.com/javase/7/docs/api/java/lang/Float.html#toString(float)
This refers to the case with some number between 1 and 10, a E, and a number representing the exponent of 10. This is still decimal and not hexadecimal.